cinn/auto_schedule/search_space/auto_gen_rule/test_helper.h

```diff
@@ -0,0 +1,260 @@
+#ifdef CINN_WITH_CUDA
+  if (type == "DeviceToHost") {
+    cudaMemcpy(dst, src, numel * sizeof(float), cudaMemcpyDeviceToHost);
+    return;
+  } else if (type == "HostToDevice") {
+    cudaMemcpy(dst, src, numel * sizeof(float), cudaMemcpyHostToDevice);
+    return;
+  }
+#endif
+  if (type == "HostToHost") {
+    for (size_t i = 0; i < numel; ++i) {
+      dst[i] = src[i];
+    }
+  } else {
+    CHECK(false) << "Unknown memory copy type";
+  }
+}
+
+void AddDataToScope(Scope* scope,
+                    const common::Target& target,
+                    float* data_ptr,
+                    int numel,
+                    std::string name,
+                    const std::vector<int>& shape) {
+  auto* var    = scope->Var<Tensor>(name);
+  auto& tensor = absl::get<Tensor>(*var);
+  CHECK(shape.size());
+  tensor->Resize(Shape(shape));
+  auto* tgt_data_ptr = tensor->mutable_data<float>(target);
+  std::string mem_cpy_type;
+#ifdef CINN_WITH_CUDA
+  mem_cpy_type = "HostToDevice";
+#else
+  mem_cpy_type = "HostToHost";
+#endif
+  MemoryCopy(data_ptr, tgt_data_ptr, numel, mem_cpy_type);
+}
+
+void InstantiateScope(Scope* scope,
+                      const common::Target& target,
+                      const std::vector<float*> input_data_ptrs,
+                      const std::vector<float*> output_data_ptrs,
+                      const std::vector<int> input_numels,
+                      const std::vector<int> output_numels,
+                      const std::vector<std::string>& input_names,
+                      const std::vector<std::string>& output_names,
+                      const std::vector<std::vector<int>>& input_shapes,
+                      const std::vector<std::vector<int>>& output_shapes) {
+  CHECK_EQ(input_names.size(), input_data_ptrs.size());
+  CHECK_EQ(input_names.size(), input_numels.size());
+  CHECK_EQ(input_names.size(), input_shapes.size());
+  CHECK_EQ(output_names.size(), output_data_ptrs.size());
+  CHECK_EQ(output_names.size(), output_numels.size());
+  CHECK_EQ(output_names.size(), output_shapes.size());
+
+  // input
+  for (int i = 0; i < input_names.size(); ++i) {
+    AddDataToScope(scope, target, input_data_ptrs[i], input_numels[i], input_names[i], input_shapes[i]);
+  }
+  // output
+  for (int i = 0; i < output_names.size(); ++i) {
+    AddDataToScope(scope, target, output_data_ptrs[i], output_numels[i], output_names[i], output_shapes[i]);
+  }
+}
+
+/* @brief: Unified signature format as the target function for comparison.
+ * @params-1: Pointers to the memory of input data, each input corresponds to a pointer.
+ * @params-2: Pointers to the memory of output data, each output corresponds to a pointer.
+ * @params-3: Shapes of the input data, each input corresponds to a std::vector<int>.
+ * @params-4: Shapes of the output data, each output corresponds to a std::vector<int>.
+ */
+using target_func_type = void (*)(const std::vector<float*>&,
```

Comment:
```suggestion
using expected_func_type = void (*)(const std::vector<float*>&,
```
建议基准数据以`expected_`为前缀，避免与后端target参数歧义，另外也体现gtest中EXPECT含义